crates/rethnet_evm/benches/state_history.rs

```diff
@@ -0,0 +1,100 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use rethnet_eth::B256;
+
+mod util;
+use util::{bench_sync_state_method, prep_no_op, SNAPSHOT_SCALES};
+
+fn bench_checkpoint(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateHistory::checkpoint()",
+        prep_no_op,
+        |mut state, _number_of_accounts| {
+            let result = state.checkpoint();
+            debug_assert!(result.is_ok());
+        },
+        &[0],
+        &[0],
```

Comment:
The number of storage slots are of interest for creating a checkpoint, as the state root is calculated

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/3921/commits/def0b581c0600ae20418a51422367108e0a9b528